### ISSUE TITLE
[bitnami/mariadb] Container not starting with root Password and MSSQL compatibility enabled

### ISSUE THREAD

**Issue by J-Insel** [bitnami/containers#16927]
### Name and Version

bitname/mariadb

### What steps will reproduce the bug?

Use the following compose file (or equivalent) to activate the mssql mode and set a root passwort.

```yaml
  mariadb:
    image: bitnami/mariadb:10.10
    environment:
      - MARIADB_ROOT_PASSWORD=test123
      - MARIADB_EXTRA_FLAGS=--sql-mode=mssql
```

The Database will initialize and then fail silently setting up authentication, resulting in a shutdown of the container. Following startup attempts (with initialized database) will result in an imidiate shutdown of the database after a failed authetication attempt of some script.

Interestingly, running the container without root password, like this:

```yaml
  mariadb:
    image: bitnami/mariadb:10.10
    environment:
      - ALLOW_EMPTY_PASSWORD=yes
      - MARIADB_EXTRA_FLAGS=--sql-mode=mssql
```

Is working fine.

### What is the expected behavior?

Beeing able to set a root passwort, when in mssql mode.

### What do you see instead?

First start:

```shell-session
$ sudo docker-compose up
Recreating mariadb_mariadb_1 ... done
Attaching to mariadb_mariadb_1
mariadb_1  | mariadb 09:21:16.18
mariadb_1  | mariadb 09:21:16.18 Welcome to the Bitnami mariadb container
mariadb_1  | mariadb 09:21:16.18 Subscribe to project updates by watching https://github.com/bitnami/containers
mariadb_1  | mariadb 09:21:16.18 Submit issues and feature requests at https://github.com/bitnami/containers/issues
mariadb_1  | mariadb 09:21:16.18
mariadb_1  | mariadb 09:21:16.18 INFO  ==> ** Starting MariaDB setup **
mariadb_1  | mariadb 09:21:16.21 INFO  ==> Validating settings in MYSQL_*/MARIADB_* env vars
mariadb_1  | mariadb 09:21:16.23 INFO  ==> Initializing mariadb database
mariadb_1  | mariadb 09:21:16.30 INFO  ==> Updating 'my.cnf' with custom configuration
mariadb_1  | mariadb 09:21:16.35 INFO  ==> Setting slow_query_log option
mariadb_1  | mariadb 09:21:16.41 INFO  ==> Setting long_query_time option
mariadb_1  | mariadb 09:21:16.43 INFO  ==> Installing database
mariadb_1  | mariadb 09:21:28.64 INFO  ==> Starting mariadb in background
mariadb_1  | 2022-12-15  9:21:28 0 [Note] InnoDB: Compressed tables use zlib 1.2.11
mariadb_1  | 2022-12-15  9:21:28 0 [Note] InnoDB: Number of transaction pools: 1
mariadb_1  | 2022-12-15  9:21:28 0 [Note] InnoDB: Using crc32 + pclmulqdq instructions
mariadb_1  | 2022-12-15  9:21:28 0 [Note] mysqld: O_TMPFILE is not supported on /opt/bitnami/mariadb/tmp (disabling future attempts)
mariadb_1  | 2022-12-15  9:21:28 0 [Note] InnoDB: Using Linux native AIO
mariadb_1  | 2022-12-15  9:21:28 0 [Note] InnoDB: Initializing buffer pool, total size = 128.000MiB, chunk size = 2.000MiB
mariadb_1  | 2022-12-15  9:21:28 0 [Note] InnoDB: Completed initialization of buffer pool
mariadb_1  | 2022-12-15  9:21:28 0 [Note] InnoDB: Buffered log writes (block size=512 bytes)
mariadb_1  | 2022-12-15  9:21:28 0 [Note] InnoDB: 128 rollback segments are active.
mariadb_1  | 2022-12-15  9:21:28 0 [Note] InnoDB: Setting file './ibtmp1' size to 12.000MiB. Physically writing the file full; Please wait ...
mariadb_1  | 2022-12-15  9:21:28 0 [Note] InnoDB: File './ibtmp1' size is now 12.000MiB.
mariadb_1  | 2022-12-15  9:21:28 0 [Note] InnoDB: log sequence number 46456; transaction id 14
mariadb_1  | 2022-12-15  9:21:28 0 [Note] InnoDB: Loading buffer pool(s) from /bitnami/mariadb/data/ib_buffer_pool
mariadb_1  | 2022-12-15  9:21:28 0 [Note] Plugin 'FEEDBACK' is disabled.
mariadb_1  | 2022-12-15  9:21:28 0 [Note] Server socket created on IP: '127.0.0.1'.
mariadb_1  | 2022-12-15  9:21:28 0 [Warning] 'user' entry 'root@5dc05cac973d' ignored in --skip-name-resolve mode.
mariadb_1  | 2022-12-15  9:21:28 0 [Warning] 'user' entry '@5dc05cac973d' ignored in --skip-name-resolve mode.
mariadb_1  | 2022-12-15  9:21:28 0 [Warning] 'proxies_priv' entry '@% root@5dc05cac973d' ignored in --skip-name-resolve mode.
mariadb_1  | 2022-12-15  9:21:29 0 [Note] InnoDB: Buffer pool(s) load completed at 221215  9:21:29
mariadb_1  | 2022-12-15  9:21:29 0 [Note] /opt/bitnami/mariadb/sbin/mysqld: ready for connections.
mariadb_1  | Version: '10.10.2-MariaDB'  socket: '/opt/bitnami/mariadb/tmp/mysql.sock'  port: 3306  Source distribution
mariadb_1  | mariadb 09:21:30.74 INFO  ==> Configuring authentication
mariadb_1  | mariadb 09:21:30.78 INFO  ==> Stopping mariadb
mariadb_1  | 2022-12-15  9:21:30 0 [Note] /opt/bitnami/mariadb/sbin/mysqld (initiated by: unknown): Normal shutdown
mariadb_1  | 2022-12-15  9:21:30 0 [Note] InnoDB: FTS optimize thread exiting.
mariadb_1  | 2022-12-15  9:21:30 0 [Note] InnoDB: Starting shutdown...
mariadb_1  | 2022-12-15  9:21:30 0 [Note] InnoDB: Dumping buffer pool(s) to /bitnami/mariadb/data/ib_buffer_pool
mariadb_1  | 2022-12-15  9:21:30 0 [Note] InnoDB: Buffer pool(s) dump completed at 221215  9:21:30
mariadb_1  | 2022-12-15  9:21:31 0 [Note] InnoDB: Removed temporary tablespace data file: "./ibtmp1"
mariadb_1  | 2022-12-15  9:21:31 0 [Note] InnoDB: Shutdown completed; log sequence number 46456; transaction id 15
mariadb_1  | 2022-12-15  9:21:31 0 [Note] /opt/bitnami/mariadb/sbin/mysqld: Shutdown complete
mariadb_1  |
mariadb_mariadb_1 exited with code 1
```

Following starts:
```shell-session
$ sudo docker-compose up
Starting mariadb_mariadb_1 ... done
Attaching to mariadb_mariadb_1
mariadb_1  | mariadb 09:24:26.90
mariadb_1  | mariadb 09:24:26.90 Welcome to the Bitnami mariadb container
mariadb_1  | mariadb 09:24:26.90 Subscribe to project updates by watching https://github.com/bitnami/containers
mariadb_1  | mariadb 09:24:26.91 Submit issues and feature requests at https://github.com/bitnami/containers/issues
mariadb_1  | mariadb 09:24:26.91
mariadb_1  | mariadb 09:24:26.91 INFO  ==> ** Starting MariaDB setup **
mariadb_1  | mariadb 09:24:27.00 INFO  ==> Validating settings in MYSQL_*/MARIADB_* env vars
mariadb_1  | mariadb 09:24:27.03 INFO  ==> Initializing mariadb database
mariadb_1  | mariadb 09:24:27.10 INFO  ==> Updating 'my.cnf' with custom configuration
mariadb_1  | mariadb 09:24:27.12 INFO  ==> Setting slow_query_log option
mariadb_1  | mariadb 09:24:27.15 INFO  ==> Setting long_query_time option
mariadb_1  | mariadb 09:24:27.36 INFO  ==> Using persisted data
mariadb_1  | mariadb 09:24:27.45 INFO  ==> Running mysql_upgrade
mariadb_1  | mariadb 09:24:27.45 INFO  ==> Starting mariadb in background
mariadb_1  | 2022-12-15  9:24:27 0 [Note] InnoDB: Compressed tables use zlib 1.2.11
mariadb_1  | 2022-12-15  9:24:27 0 [Note] InnoDB: Number of transaction pools: 1
mariadb_1  | 2022-12-15  9:24:27 0 [Note] InnoDB: Using crc32 + pclmulqdq instructions
mariadb_1  | 2022-12-15  9:24:27 0 [Note] mysqld: O_TMPFILE is not supported on /opt/bitnami/mariadb/tmp (disabling future attempts)
mariadb_1  | 2022-12-15  9:24:27 0 [Note] InnoDB: Using Linux native AIO
mariadb_1  | 2022-12-15  9:24:27 0 [Note] InnoDB: Initializing buffer pool, total size = 128.000MiB, chunk size = 2.000MiB
mariadb_1  | 2022-12-15  9:24:28 0 [Note] InnoDB: Completed initialization of buffer pool
mariadb_1  | 2022-12-15  9:24:28 0 [Note] InnoDB: Buffered log writes (block size=512 bytes)
mariadb_1  | 2022-12-15  9:24:28 0 [Note] InnoDB: 128 rollback segments are active.
mariadb_1  | 2022-12-15  9:24:28 0 [Note] InnoDB: Setting file './ibtmp1' size to 12.000MiB. Physically writing the file full; Please wait ...
mariadb_1  | 2022-12-15  9:24:28 0 [Note] InnoDB: File './ibtmp1' size is now 12.000MiB.
mariadb_1  | 2022-12-15  9:24:28 0 [Note] InnoDB: log sequence number 46456; transaction id 14
mariadb_1  | 2022-12-15  9:24:28 0 [Note] InnoDB: Loading buffer pool(s) from /bitnami/mariadb/data/ib_buffer_pool
mariadb_1  | 2022-12-15  9:24:28 0 [Note] Plugin 'FEEDBACK' is disabled.
mariadb_1  | 2022-12-15  9:24:28 0 [Note] Server socket created on IP: '127.0.0.1'.
mariadb_1  | 2022-12-15  9:24:28 0 [Note] InnoDB: Buffer pool(s) load completed at 221215  9:24:28
mariadb_1  | 2022-12-15  9:24:28 0 [Warning] 'proxies_priv' entry '@% root@5dc05cac973d' ignored in --skip-name-resolve mode.
mariadb_1  | 2022-12-15  9:24:28 0 [Note] /opt/bitnami/mariadb/sbin/mysqld: ready for connections.
mariadb_1  | Version: '10.10.2-MariaDB'  socket: '/opt/bitnami/mariadb/tmp/mysql.sock'  port: 3306  Source distribution
mariadb_1  | 2022-12-15  9:24:29 3 [Warning] Access denied for user 'root'@'localhost' (using password: YES)
mariadb_1  | mariadb 09:24:29.68 INFO  ==> Stopping mariadb
mariadb_1  | 2022-12-15  9:24:29 0 [Note] /opt/bitnami/mariadb/sbin/mysqld (initiated by: unknown): Normal shutdown
mariadb_1  | 2022-12-15  9:24:29 0 [Note] InnoDB: FTS optimize thread exiting.
mariadb_1  | 2022-12-15  9:24:29 0 [Note] InnoDB: Starting shutdown...
mariadb_1  | 2022-12-15  9:24:29 0 [Note] InnoDB: Dumping buffer pool(s) to /bitnami/mariadb/data/ib_buffer_pool
mariadb_1  | 2022-12-15  9:24:29 0 [Note] InnoDB: Buffer pool(s) dump completed at 221215  9:24:29
mariadb_1  | 2022-12-15  9:24:29 0 [Note] InnoDB: Removed temporary tablespace data file: "./ibtmp1"
mariadb_1  | 2022-12-15  9:24:29 0 [Note] InnoDB: Shutdown completed; log sequence number 46456; transaction id 15
mariadb_1  | 2022-12-15  9:24:29 0 [Note] /opt/bitnami/mariadb/sbin/mysqld: Shutdown complete
mariadb_1  |
mariadb_mariadb_1 exited with code 1
```

**Comment by msaustral** [comment]
Hi we have 2 days try to implement Mysql/Mariadb/Galera any of them and getting the same error on the replica

[Access denied for user 'root'@'localhost](message: "Readiness probe failed: \amysqladmin: connect to server at 'localhost' failed\nerror: 'Access denied for user 'root'@'localhost' (using password: YES)')

We got the helm and only apply namespace and change standalone by replication and the secondary never gets up

Any idea??

AKS

**Comment by mdhont** [comment]
Could you add BITNAMI_DEBUG=1 to the docker-compose.yaml environment and run the docker compose command?

```
    environment:
      - MARIADB_EXTRA_FLAGS=--sql-mode=mssql
      - BITNAMI_DEBUG=1
      - MARIADB_ROOT_PASSWORD=test123
```

**Comment by msaustral** [comment]
Hi we are using helm on kubernetes,

we download the value file from here: https://github.com/bitnami/charts/blob/main/bitnami/mysql/values.yaml

Change:

the file name to mysql.yaml
line 32 with namespace
line 96 to replication
line 103/118/125 password (same for all)
line 115 name of the user
line 452/826 to LoadBalancer
line 1009 set to true metrics
helm install arrow-prod -f mysql.yaml azure-marketplace/mysql

After 14 min:

the primary goes up, you can connect to it and start to work with it
the secondary remains down:
message: "Startup probe failed: mysqladmin: [Warning] Using a password on the command line interface can be insecure.\n\amysqladmin: connect to server at 'localhost' failed\nerror: 'Access denied for user 'root'@'localhost' (using password: YES)'

This happens with Mysql, Madiadb and Galera chart

**Comment by jackblk** [comment]
I have the same issue.

Here's the log with `BITNAMI_DEBUG=1`

```
$ docker logs -f mariadb 
mariadb 11:43:34.29 
mariadb 11:43:34.30 Welcome to the Bitnami mariadb container
mariadb 11:43:34.30 Subscribe to project updates by watching https://github.com/bitnami/containers
mariadb 11:43:34.31 Submit issues and feature requests at https://github.com/bitnami/containers/issues
mariadb 11:43:34.31 
mariadb 11:43:34.32 INFO  ==> ** Starting MariaDB setup **
mariadb 11:43:34.38 INFO  ==> Validating settings in MYSQL_*/MARIADB_* env vars
mariadb 11:43:34.41 INFO  ==> Initializing mariadb database
mariadb 11:43:34.42 DEBUG ==> Ensuring expected directories/files exist
mariadb 11:43:34.44 INFO  ==> Updating 'my.cnf' with custom configuration
mariadb 11:43:34.45 INFO  ==> Setting user option
mariadb 11:43:34.46 DEBUG ==> Setting user to 'bn_wordpress' in mariadb configuration file /opt/bitnami/mariadb/conf/my.cnf
mariadb 11:43:34.49 INFO  ==> Setting slow_query_log option
mariadb 11:43:34.49 DEBUG ==> Setting slow_query_log to '0' in mariadb configuration file /opt/bitnami/mariadb/conf/my.cnf
mariadb 11:43:34.50 INFO  ==> Setting long_query_time option
mariadb 11:43:34.50 DEBUG ==> Setting long_query_time to '10.0' in mariadb configuration file /opt/bitnami/mariadb/conf/my.cnf
mariadb 11:43:34.51 INFO  ==> Using persisted data
mariadb 11:43:34.57 INFO  ==> Running mysql_upgrade
mariadb 11:43:34.58 INFO  ==> Starting mariadb in background
2022-12-25 11:43:34 0 [Note] /opt/bitnami/mariadb/sbin/mysqld (server 10.9.4-MariaDB) starting as process 72 ...
2022-12-25 11:43:34 0 [Note] InnoDB: Compressed tables use zlib 1.2.11
2022-12-25 11:43:34 0 [Note] InnoDB: Using transactional memory
2022-12-25 11:43:34 0 [Note] InnoDB: Number of transaction pools: 1
2022-12-25 11:43:34 0 [Note] InnoDB: Using crc32 + pclmulqdq instructions
2022-12-25 11:43:34 0 [Note] mysqld: O_TMPFILE is not supported on /opt/bitnami/mariadb/tmp (disabling future attempts)
2022-12-25 11:43:34 0 [Note] InnoDB: Using Linux native AIO
2022-12-25 11:43:34 0 [Note] InnoDB: Initializing buffer pool, total size = 128.000MiB, chunk size = 2.000MiB
2022-12-25 11:43:34 0 [Note] InnoDB: Completed initialization of buffer pool
2022-12-25 11:43:34 0 [Note] InnoDB: File system buffers for log disabled (block size=4096 bytes)
2022-12-25 11:43:34 0 [Note] InnoDB: 128 rollback segments are active.
2022-12-25 11:43:34 0 [Note] InnoDB: Setting file './ibtmp1' size to 12.000MiB. Physically writing the file full; Please wait ...
2022-12-25 11:43:34 0 [Note] InnoDB: File './ibtmp1' size is now 12.000MiB.
2022-12-25 11:43:34 0 [Note] InnoDB: log sequence number 47867740; transaction id 28273
2022-12-25 11:43:34 0 [Note] InnoDB: Loading buffer pool(s) from /bitnami/mariadb/data/ib_buffer_pool
2022-12-25 11:43:34 0 [Note] Plugin 'FEEDBACK' is disabled.
2022-12-25 11:43:34 0 [Note] Server socket created on IP: '127.0.0.1'.
2022-12-25 11:43:34 0 [Note] InnoDB: Buffer pool(s) load completed at 221225 11:43:34
2022-12-25 11:43:34 0 [Warning] 'proxies_priv' entry '@% root@a95d405e1aee' ignored in --skip-name-resolve mode.
2022-12-25 11:43:34 0 [Note] /opt/bitnami/mariadb/sbin/mysqld: ready for connections.
Version: '10.9.4-MariaDB'  socket: '/opt/bitnami/mariadb/tmp/mysql.sock'  port: 3306  Source distribution
Phase 1/7: Checking and upgrading mysql database
Processing databases
mysql
mysql.column_stats
Error    : Incorrect file format 'column_stats'
error    : Corrupt
mysql.columns_priv                                 OK
mysql.db                                           OK
mysql.event                                        OK
mysql.func                                         OK
mysql.global_priv                                  OK
mysql.gtid_slave_pos                               OK
mysql.help_category                                OK
mysql.help_keyword
Error    : Incorrect file format 'help_keyword'
error    : Corrupt
mysql.help_relation
Error    : Incorrect file format 'help_relation'
error    : Corrupt
mysql.help_topic
Error    : Incorrect file format 'help_topic'
error    : Corrupt
mysql.index_stats
Error    : Incorrect file format 'index_stats'
error    : Corrupt
mysql.innodb_index_stats                           OK
mysql.innodb_table_stats                           OK
mysql.plugin                                       OK
mysql.proc                                         OK
mysql.procs_priv                                   OK
mysql.proxies_priv                                 OK
mysql.roles_mapping                                OK
mysql.servers                                      OK
mysql.table_stats
Error    : Incorrect file format 'table_stats'
error    : Corrupt
mysql.tables_priv                                  OK
mysql.time_zone                                    OK
mysql.time_zone_leap_second                        OK
mysql.time_zone_name                               OK
mysql.time_zone_transition                         OK
mysql.time_zone_transition_type                    OK
mysql.transaction_registry                         OK

Repairing tables
mysql.column_stats
Error    : Incorrect file format 'column_stats'
error    : Corrupt
mysql.help_keyword
Error    : Incorrect file format 'help_keyword'
error    : Corrupt
mysql.help_relation
Error    : Incorrect file format 'help_relation'
error    : Corrupt
mysql.help_topic
Error    : Incorrect file format 'help_topic'
error    : Corrupt
mysql.index_stats
Error    : Incorrect file format 'index_stats'
error    : Corrupt
mysql.table_stats
Error    : Incorrect file format 'table_stats'
error    : Corrupt
Phase 2/7: Installing used storage engines... Skipped
Phase 3/7: Fixing views
mysql.user                                         OK
sys.host_summary                                   OK
sys.host_summary_by_file_io                        OK
sys.host_summary_by_file_io_type                   OK
sys.host_summary_by_stages                         OK
sys.host_summary_by_statement_latency              OK
sys.host_summary_by_statement_type                 OK
sys.innodb_buffer_stats_by_schema                  OK
sys.innodb_buffer_stats_by_table                   OK
sys.innodb_lock_waits                              OK
sys.io_by_thread_by_latency                        OK
sys.io_global_by_file_by_bytes                     OK
sys.io_global_by_file_by_latency                   OK
sys.io_global_by_wait_by_bytes                     OK
sys.io_global_by_wait_by_latency                   OK
sys.latest_file_io                                 OK
sys.memory_by_host_by_current_bytes                OK
sys.memory_by_thread_by_current_bytes              OK
sys.memory_by_user_by_current_bytes                OK
sys.memory_global_by_current_bytes                 OK
sys.memory_global_total                            OK
sys.metrics                                        OK
sys.processlist                                    OK
sys.ps_check_lost_instrumentation                  OK
sys.schema_auto_increment_columns                  OK
sys.schema_index_statistics                        OK
sys.schema_object_overview                         OK
sys.schema_redundant_indexes                       OK
sys.schema_table_lock_waits                        OK
sys.schema_table_statistics                        OK
sys.schema_table_statistics_with_buffer            OK
sys.schema_tables_with_full_table_scans            OK
sys.schema_unused_indexes                          OK
sys.session                                        OK
sys.session_ssl_status                             OK
sys.statement_analysis                             OK
sys.statements_with_errors_or_warnings             OK
sys.statements_with_full_table_scans               OK
sys.statements_with_runtimes_in_95th_percentile    OK
sys.statements_with_sorting                        OK
sys.statements_with_temp_tables                    OK
sys.user_summary                                   OK
sys.user_summary_by_file_io                        OK
sys.user_summary_by_file_io_type                   OK
sys.user_summary_by_stages                         OK
sys.user_summary_by_statement_latency              OK
sys.user_summary_by_statement_type                 OK
sys.version                                        OK
sys.wait_classes_global_by_avg_latency             OK
sys.wait_classes_global_by_latency                 OK
sys.waits_by_host_by_latency                       OK
sys.waits_by_user_by_latency                       OK
sys.waits_global_by_latency                        OK
sys.x$host_summary                                 OK
sys.x$host_summary_by_file_io                      OK
sys.x$host_summary_by_file_io_type                 OK
sys.x$host_summary_by_stages                       OK
sys.x$host_summary_by_statement_latency            OK
sys.x$host_summary_by_statement_type               OK
sys.x$innodb_buffer_stats_by_schema                OK
sys.x$innodb_buffer_stats_by_table                 OK
sys.x$innodb_lock_waits                            OK
sys.x$io_by_thread_by_latency                      OK
sys.x$io_global_by_file_by_bytes                   OK
sys.x$io_global_by_file_by_latency                 OK
sys.x$io_global_by_wait_by_bytes                   OK
sys.x$io_global_by_wait_by_latency                 OK
sys.x$latest_file_io                               OK
sys.x$memory_by_host_by_current_bytes              OK
sys.x$memory_by_thread_by_current_bytes            OK
sys.x$memory_by_user_by_current_bytes              OK
sys.x$memory_global_by_current_bytes               OK
sys.x$memory_global_total                          OK
sys.x$processlist                                  OK
sys.x$ps_digest_95th_percentile_by_avg_us          OK
sys.x$ps_digest_avg_latency_distribution           OK
sys.x$ps_schema_table_statistics_io                OK
sys.x$schema_flattened_keys                        OK
sys.x$schema_index_statistics                      OK
sys.x$schema_table_lock_waits                      OK
sys.x$schema_table_statistics                      OK
sys.x$schema_table_statistics_with_buffer          OK
sys.x$schema_tables_with_full_table_scans          OK
sys.x$session                                      OK
sys.x$statement_analysis                           OK
sys.x$statements_with_errors_or_warnings           OK
sys.x$statements_with_full_table_scans             OK
sys.x$statements_with_runtimes_in_95th_percentile  OK
sys.x$statements_with_sorting                      OK
sys.x$statements_with_temp_tables                  OK
sys.x$user_summary                                 OK
sys.x$user_summary_by_file_io                      OK
sys.x$user_summary_by_file_io_type                 OK
sys.x$user_summary_by_stages                       OK
sys.x$user_summary_by_statement_latency            OK
sys.x$user_summary_by_statement_type               OK
sys.x$wait_classes_global_by_avg_latency           OK
sys.x$wait_classes_global_by_latency               OK
sys.x$waits_by_host_by_latency                     OK
sys.x$waits_by_user_by_latency                     OK
sys.x$waits_global_by_latency                      OK
Phase 4/7: Running 'mysql_fix_privilege_tables'
ERROR 130 (HY000) at line 5: Incorrect file format 'column_stats'
ERROR 130 (HY000) at line 31: Incorrect file format 'help_topic'
ERROR 130 (HY000) at line 33: Incorrect file format 'help_relation'
ERROR 130 (HY000) at line 34: Incorrect file format 'help_keyword'
ERROR 130 (HY000) at line 35: Incorrect file format 'table_stats'
ERROR 130 (HY000) at line 36: Incorrect file format 'index_stats'
ERROR 130 (HY000) at line 58: Incorrect file format 'column_stats'
ERROR 130 (HY000) at line 77: Incorrect file format 'column_stats'
ERROR 130 (HY000) at line 146: Incorrect file format 'column_stats'
ERROR 130 (HY000) at line 164: Incorrect file format 'column_stats'
ERROR 130 (HY000) at line 237: Incorrect file format 'column_stats'
ERROR 130 (HY000) at line 299: Incorrect file format 'column_stats'
ERROR 130 (HY000) at line 366: Incorrect file format 'index_stats'
ERROR 130 (HY000) at line 505: Incorrect file format 'help_topic'
FATAL ERROR: Upgrade failed
mariadb 11:43:39.72 INFO  ==> Stopping mariadb
2022-12-25 11:43:39 0 [Note] /opt/bitnami/mariadb/sbin/mysqld (initiated by: unknown): Normal shutdown
2022-12-25 11:43:39 0 [Note] InnoDB: FTS optimize thread exiting.
2022-12-25 11:43:39 0 [Note] InnoDB: Starting shutdown...
2022-12-25 11:43:39 0 [Note] InnoDB: Dumping buffer pool(s) to /bitnami/mariadb/data/ib_buffer_pool
2022-12-25 11:43:39 0 [Note] InnoDB: Buffer pool(s) dump completed at 221225 11:43:39
2022-12-25 11:43:39 0 [Note] InnoDB: Removed temporary tablespace data file: "./ibtmp1"
2022-12-25 11:43:39 0 [Note] InnoDB: Shutdown completed; log sequence number 47867740; transaction id 28277
2022-12-25 11:43:39 0 [Note] /opt/bitnami/mariadb/sbin/mysqld: Shutdown complete

mariadb 11:43:40.73 DEBUG ==> Waiting for mariadb to unlock db files
```

**Comment by jackblk** [comment]
I'm able to fix my issue. I created a new container, mount the data volume, rewrite entrypoint with bash shell, run the mariadb in background with `/opt/bitnami/scripts/mariadb/run.sh &`, login using `mysql -u root -p`, then repair my tables:

```
  MariaDB> use mysql;
  MariaDB> repair table user use_frm;
  MariaDB> check table user use_frm;;
  MariaDB> exit;
```

**Comment by msaustral** [comment]
Hi, on helm values is not an option for mount point.

Any manual fix is viable, when the container scales up, it won't work

**Comment by J-Insel** [comment]
This is the log output with debug enabled:

```
mariadb_1  | mariadb 09:06:33.05
mariadb_1  | mariadb 09:06:33.05 Welcome to the Bitnami mariadb container
mariadb_1  | mariadb 09:06:33.05 Subscribe to project updates by watching https://github.com/bitnami/containers
mariadb_1  | mariadb 09:06:33.05 Submit issues and feature requests at https://github.com/bitnami/containers/issues
mariadb_1  | mariadb 09:06:33.05
mariadb_1  | mariadb 09:06:33.06 INFO  ==> ** Starting MariaDB setup **
mariadb_1  | mariadb 09:06:33.08 INFO  ==> Validating settings in MYSQL_*/MARIADB_* env vars
mariadb_1  | mariadb 09:06:33.11 INFO  ==> Initializing mariadb database
mariadb_1  | mariadb 09:06:33.11 DEBUG ==> Ensuring expected directories/files exist
mariadb_1  | mariadb 09:06:33.14 INFO  ==> Updating 'my.cnf' with custom configuration
mariadb_1  | mariadb 09:06:33.14 INFO  ==> Setting slow_query_log option
mariadb_1  | mariadb 09:06:33.15 DEBUG ==> Setting slow_query_log to '0' in mariadb configuration file /opt/bitnami/mariadb/conf/my.cnf
mariadb_1  | mariadb 09:06:33.16 INFO  ==> Setting long_query_time option
mariadb_1  | mariadb 09:06:33.16 DEBUG ==> Setting long_query_time to '10.0' in mariadb configuration file /opt/bitnami/mariadb/conf/my.cnf
mariadb_1  | mariadb 09:06:33.18 DEBUG ==> Cleaning data directory to ensure successfully initialization
mariadb_1  | mariadb 09:06:33.19 INFO  ==> Installing database
mariadb_1  | Installing MariaDB/MySQL system tables in '/bitnami/mariadb/data' ...
mariadb_1  | OK
mariadb_1  |
mariadb_1  | To start mysqld at boot time you have to copy
mariadb_1  | support-files/mysql.server to the right place for your system
mariadb_1  |
mariadb_1  |
mariadb_1  | PLEASE REMEMBER TO SET A PASSWORD FOR THE MariaDB root USER !
mariadb_1  | To do so, start the server, then issue the following command:
mariadb_1  |
mariadb_1  | '/opt/bitnami/mariadb/bin/mysql_secure_installation'
mariadb_1  |
mariadb_1  | which will also give you the option of removing the test
mariadb_1  | databases and anonymous user created by default.  This is
mariadb_1  | strongly recommended for production servers.
mariadb_1  |
mariadb_1  | See the MariaDB Knowledgebase at https://mariadb.com/kb
mariadb_1  |
mariadb_1  | You can start the MariaDB daemon with:
mariadb_1  | cd '/opt/bitnami/mariadb' ; /opt/bitnami/mariadb/bin/mysqld_safe --datadir='/bitnami/mariadb/data'
mariadb_1  |
mariadb_1  | You can test the MariaDB daemon with mysql-test-run.pl
mariadb_1  | cd '/opt/bitnami/mariadb/mysql-test' ; perl mysql-test-run.pl
mariadb_1  |
mariadb_1  | Please report any problems at https://mariadb.org/jira
mariadb_1  |
mariadb_1  | The latest information about MariaDB is available at https://mariadb.org/.
mariadb_1  |
mariadb_1  | Consider joining MariaDB's strong and vibrant community:
mariadb_1  | https://mariadb.org/get-involved/
mariadb_1  |
mariadb_1  | mariadb 09:06:36.76 INFO  ==> Starting mariadb in background
mariadb_1  | 2022-12-27  9:06:36 0 [Note] /opt/bitnami/mariadb/sbin/mysqld (server 10.10.2-MariaDB) starting as process 100 ...
mariadb_1  | 2022-12-27  9:06:36 0 [Note] InnoDB: Compressed tables use zlib 1.2.11
mariadb_1  | 2022-12-27  9:06:36 0 [Note] InnoDB: Number of transaction pools: 1
mariadb_1  | 2022-12-27  9:06:36 0 [Note] InnoDB: Using crc32 + pclmulqdq instructions
mariadb_1  | 2022-12-27  9:06:36 0 [Note] mysqld: O_TMPFILE is not supported on /opt/bitnami/mariadb/tmp (disabling future attempts)
mariadb_1  | 2022-12-27  9:06:36 0 [Note] InnoDB: Using Linux native AIO
mariadb_1  | 2022-12-27  9:06:36 0 [Note] InnoDB: Initializing buffer pool, total size = 128.000MiB, chunk size = 2.000MiB
mariadb_1  | 2022-12-27  9:06:36 0 [Note] InnoDB: Completed initialization of buffer pool
mariadb_1  | 2022-12-27  9:06:36 0 [Note] InnoDB: Buffered log writes (block size=512 bytes)
mariadb_1  | 2022-12-27  9:06:36 0 [Note] InnoDB: 128 rollback segments are active.
mariadb_1  | 2022-12-27  9:06:36 0 [Note] InnoDB: Setting file './ibtmp1' size to 12.000MiB. Physically writing the file full; Please wait ...
mariadb_1  | 2022-12-27  9:06:36 0 [Note] InnoDB: File './ibtmp1' size is now 12.000MiB.
mariadb_1  | 2022-12-27  9:06:36 0 [Note] InnoDB: log sequence number 46590; transaction id 14
mariadb_1  | 2022-12-27  9:06:36 0 [Note] InnoDB: Loading buffer pool(s) from /bitnami/mariadb/data/ib_buffer_pool
mariadb_1  | 2022-12-27  9:06:36 0 [Note] Plugin 'FEEDBACK' is disabled.
mariadb_1  | 2022-12-27  9:06:36 0 [Note] InnoDB: Buffer pool(s) load completed at 221227  9:06:36
mariadb_1  | 2022-12-27  9:06:36 0 [Note] Server socket created on IP: '127.0.0.1'.
mariadb_1  | 2022-12-27  9:06:36 0 [Warning] 'user' entry 'root@00d823d58e28' ignored in --skip-name-resolve mode.
mariadb_1  | 2022-12-27  9:06:36 0 [Warning] 'user' entry '@00d823d58e28' ignored in --skip-name-resolve mode.
mariadb_1  | 2022-12-27  9:06:36 0 [Warning] 'proxies_priv' entry '@% root@00d823d58e28' ignored in --skip-name-resolve mode.
mariadb_1  | 2022-12-27  9:06:36 0 [Note] /opt/bitnami/mariadb/sbin/mysqld: ready for connections.
mariadb_1  | Version: '10.10.2-MariaDB'  socket: '/opt/bitnami/mariadb/tmp/mysql.sock'  port: 3306  Source distribution
mariadb_1  | 1
mariadb_1  | mariadb 09:06:38.79 INFO  ==> Configuring authentication
mariadb_1  | mariadb 09:06:38.80 DEBUG ==> Configuring root user credentials
mariadb_1  | ERROR 1064 (42000) at line 5: You have an error in your SQL syntax; check the manual that corresponds to your MariaDB server version for the right syntax to use near '"test123"' at line 1
mariadb_1  | mariadb 09:06:38.82 INFO  ==> Stopping mariadb
mariadb_1  | 2022-12-27  9:06:38 0 [Note] /opt/bitnami/mariadb/sbin/mysqld (initiated by: unknown): Normal shutdown
mariadb_1  | 2022-12-27  9:06:38 0 [Note] InnoDB: FTS optimize thread exiting.
mariadb_1  | 2022-12-27  9:06:38 0 [Note] InnoDB: Starting shutdown...
mariadb_1  | 2022-12-27  9:06:38 0 [Note] InnoDB: Dumping buffer pool(s) to /bitnami/mariadb/data/ib_buffer_pool
mariadb_1  | 2022-12-27  9:06:38 0 [Note] InnoDB: Buffer pool(s) dump completed at 221227  9:06:38
mariadb_1  | 2022-12-27  9:06:39 0 [Note] InnoDB: Removed temporary tablespace data file: "./ibtmp1"
mariadb_1  | 2022-12-27  9:06:39 0 [Note] InnoDB: Shutdown completed; log sequence number 46590; transaction id 15
mariadb_1  | 2022-12-27  9:06:39 0 [Note] /opt/bitnami/mariadb/sbin/mysqld: Shutdown complete
mariadb_1  |
mariadb_1  | mariadb 09:06:39.83 DEBUG ==> Waiting for mariadb to unlock db files
```

Btw. setting a root passwort is working perfectly fine without mssql compatiblity mode. Unfortunately we need it for an application that got adopted and is not yet rewriten for mysql syntax.

**Comment by msaustral** [comment]
Hi, any news? any solution??

**Comment by mdhont** [comment]
I've reproduced the error, I have created a task to fix it and I will update this thread with further information.

**Comment by msaustral** [comment]
Hi @mdhont thank you, how can we know when the fix is ready?

**Comment by mdhont** [comment]
I will notify you in this thread.

**Comment by msaustral** [comment]
thank you

**Comment by andresbono** [comment]
Hi @J-Insel, could you try the following?

`docker-compose.yml`
```yaml
  mariadb:
    image: docker.io/bitnami/mariadb:10.10
    environment:
      - MARIADB_ROOT_USER=root
      - MARIADB_ROOT_PASSWORD=test123
    command: ["/opt/bitnami/scripts/mariadb/run.sh", "--sql-mode=mssql"]
```

You can check if the MSSQL flag is set with these commands:

```shell
$ docker-compose up -d
$ docker-compose exec mariadb mysql -u root -ptest123 -e "SELECT @@SQL_MODE LIKE '%MSSQL%'"
+---------------------------+
| @@SQL_MODE LIKE '%MSSQL%' |
+---------------------------+
|                         1 |
+---------------------------+
```

---

This will activate the MSSQL mode only after the initialization is complete, without affecting the logic that configures the root user (and other steps):

https://github.com/bitnami/containers/blob/8390c743772ed86588fa319d54bdd2c6ced41c77/bitnami/mariadb/10.10/debian-11/rootfs/opt/bitnami/scripts/libmariadb.sh#L944-L952

It is expected to have issues if the SQL code written in the initialization scripts is not compatible with the SQL mode specified when starting the server.

Let us know if it works for you.

**Comment by msaustral** [comment]
Hi we are using Helm chart on AKS, not docker, will it work?

**Comment by andresbono** [comment]
@msaustral: there shouldn't be any problem if you use the helm chart on AKS. Please, let's keep this issue to talk about the MSSQL mode incompatibility. Thanks.

**Comment by msaustral** [comment]
The issue that we had was that the second replica do not start and has the error: Container not starting with root Password

**Comment by mdhont** [comment]
@msaustral, could you open a separate issue for the second replica not starting? I'm proceeding to close this issue.
